module.nix

```diff
@@ -1,96 +1,128 @@
+        example = lib.literalExpression "\${pkgs.qemu_kvm}/bin/qemu-system-aarch64";
+      };
+
+      pkgs = lib.mkOption {
+        type = lib.types.attrs;
+        description = ''
+          the pkgs instance used when building disk images via make-disk-image.nix.
+          Useful when the config's kernel won't boot in the image-builder.
+        '';
+        default = pkgs;
+        defaultText = lib.literalExpression "pkgs";
+        example = lib.literalExpression "pkgs";
+      };
+
+      kernelPackages = lib.mkOption {
+        type = lib.types.attrs;
+        description = ''
+          the kernel used when building disk images via make-disk-image.nix.
+          Useful when the config's kernel won't boot in the image-builder.
+        '';
+        default = config.boot.kernelPackages;
+        defaultText = lib.literalExpression "config.boot.kernelPackages";
+        example = lib.literalExpression "pkgs.linuxPackages_testing";
+      };
+
+      extraRootModules = lib.mkOption {
+        type = lib.types.listOf lib.types.str;
+        description = ''
+          extra kernel modules to pass to the vmTools.runCommand invocation in the make-disk-image.nix builder
+        '';
+        default = [ ];
+        example = [ "bcachefs" ];
+      };
+
+      extraPostVM = lib.mkOption {
+        type = lib.types.str;
+        description = ''
+          extra shell code to execute once the disk image(s) have been succesfully created and moved to $out
+        '';
+        default = "";
+        example = lib.literalExpression ''
+          ''${pkgs.zstd}/bin/zstd --compress $out/*raw
+          rm $out/*raw
+        '';
+      };
+
+      extraDependencies = lib.mkOption {
+        type = lib.types.listOf lib.types.package;
+        description = ''
+          list of extra packages to make available in the make-disk-image.nix VM builder, an example might be f2fs-tools
+        '';
+        default = [ ];
+      };
+
+      name = lib.mkOption {
+        type = lib.types.str;
+        description = "name for the disk images";
+        default = "${config.networking.hostName}-disko-images";
+      };
+
+      copyNixStore = lib.mkOption {
+        type = lib.types.bool;
+        description = "whether to copy the nix store into the disk images we just created";
+        default = true;
+      };
+
+      extraConfig = lib.mkOption {
+        description = ''
+          Extra NixOS config for your test. Can be used to specify a different luks key for tests.
+          A dummy key is in /tmp/secret.key
+        '';
+        default = { };
+      };
+
+      imageFormat = lib.mkOption {
+        type = lib.types.enum [ "raw" "qcow2" ];
+        description = "QEMU image format to use for the disk images";
+        default = "raw";
+      };
     };
+
     memSize = lib.mkOption {
       type = lib.types.int;
       description = ''
         size of the memory passed to runInLinuxVM, in megabytes
       '';
       default = 1024;
     };
```

Comment:
should memSize have been moved, too?

Reply:
I didn't move it because it's being shared by interactive VM: https://github.com/nix-community/disko/blob/51e3a7e51279fedfb6669a00d21dc5936c78a6ce/lib/interactive-vm.nix#L59

Maybe the comment needs updating